packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FLTWFInstanceManager.m

```diff
@@ -0,0 +1,100 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "FLTWFInstanceManager.h"
+
+@interface FLTThreadSafeMapTable<KeyType, ObjectType> : NSObject
++ (nonnull FLTThreadSafeMapTable *)strongToStrongMapTable;
+- (void)setObject:(nonnull ObjectType)object forKey:(nonnull KeyType)key;
+- (void)removeObjectForKey:(nonnull KeyType)key;
+- (nullable ObjectType)objectForKey:(nonnull KeyType)key;
+@end
+
+@implementation FLTThreadSafeMapTable {
+  NSMapTable<id, id> *_table;
+  dispatch_queue_t _lockQueue;
+}
+
++ (FLTThreadSafeMapTable *)strongToStrongMapTable {
+  return [[FLTThreadSafeMapTable alloc] initWithTable:[NSMapTable strongToStrongObjectsMapTable]];
+}
+
+- (nonnull instancetype)initWithTable:(NSMapTable *)table {
+  self = [super init];
+  if (self) {
+    _lockQueue = dispatch_queue_create("FLTThreadSafeMapTable", DISPATCH_QUEUE_SERIAL);
+    _table = table;
+  }
+  return self;
+}
+
+- (void)setObject:(nonnull id)object forKey:(nonnull id)key {
+  if (key && object) {
+    dispatch_async(_lockQueue, ^{
+      [self->_table setObject:object forKey:key];
+    });
+  }
+}
+
+- (void)removeObjectForKey:(nonnull id)key {
+  if (key != nil) {
+    dispatch_async(_lockQueue, ^{
+      [self->_table removeObjectForKey:key];
+    });
+  }
+}
+
+- (nullable id)objectForKey:(nonnull id)key {
+  id __block object = nil;
+  dispatch_sync(_lockQueue, ^{
+    object = [self->_table objectForKey:key];
+  });
+  return object;
+}
+@end
+
+@implementation FLTWFInstanceManager {
+  FLTThreadSafeMapTable<NSNumber *, NSObject *> *_instanceIDsToInstances;
```

Comment:
Use properties here, and call them `identifiersToInstances` and `instancesToIdentifiers` (not repeating "instance" makes parsing the direction easier).